crates/search/src/buffer_search.rs

```diff
@@ -556,6 +556,9 @@ impl BufferSearchBar {
             active_editor.toggle_filtered_search_ranges(false, cx);
             let handle = active_editor.focus_handle(cx);
             cx.focus(&handle);
+            cx.on_next_frame(|_, cx| {
```

Comment:
I'm noticing that a lot of these are on next frame, rather than current frame. What's going on there?

Reply:
This one I placed here because we were not calling `selection_did_change` when leaving the buffer search.